src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useMemo, useState } from 'react';
+import { useTodos } from '../../TodosContext';
+import { TodoItem } from '../TodoItem';
+import { TodosFilter } from '../TodosFilter';
+import { Status } from '../../utils/Status';
+
+export const TodoList = () => {
+  const { todos, setTodos } = useTodos();
+
+  const [filterParam, setFilterParam] = useState(Status.All);
+
+  const handleToggleAll = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodos(prev => prev.map(todo => ({
+      ...todo, completed: event.target.checked,
+    })));
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterParam) {
+        case (Status.All):
+          return todo;
+        case (Status.Active):
+          return !todo.completed;
+        case (Status.Completed):
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }, [filterParam, todos]);
```

Comment:
https://github.com/mate-academy/react_todo-app/pull/679#discussion_r1327520587 - not fully fixed